middlewarerule.js

```diff
@@ -222,8 +222,15 @@ MiddlewareRule.prototype.getCollection = function(params, data, req) {
 	}
 
 	var nonSpecialParams = Object.keys(filteredParams).filter(function(key) {
+		// Exclude special params.
 		return specialParams.indexOf(key) === -1;
-	});
+	}).filter(function(key) {
+		// Exclude path params.
```

Comment:
This code excludes the path params from the collection filtering -- only query parameters should be considered.